### PR TITLE
        update the AD settings for opensearch.

### PR DESCRIPTION
**Pull request by spbjss**
Signed-off-by: Alex <pengsun@amazon.com>

### Description
Rename plugin settings from opendistro to opensearch.
 
### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/40
 
### Check List
- [Y] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/ad/transport/ADBatchAnomalyResultTransportActionTests.java
Comment (kaituo) [inline src/test/java/org/opensearch/ad/transport/ADBatchAnomalyResultTransportActionTests.java:140]: This will make our tests slower and slower when adding more tests.  We want to avoid even though the execute finishes quickly and we still sleep 20 seconds.  You can use a CountDownLatch to wait while the client().execute is running.  See an example in EntityProfileRunnerTests.stateTestTemplate

Comment (spbjss) [reply src/test/java/org/opensearch/ad/transport/ADBatchAnomalyResultTransportActionTests.java:140]: Thank you very much for the suggestion! 
I have checked with Yaliang, he is still working on the batching change and need more time to optimize the unit tests.
I'm planning to update these unit tests after Yaliang finished his PRs.

Comment (ylwu-amzn) [reply src/test/java/org/opensearch/ad/transport/ADBatchAnomalyResultTransportActionTests.java:140]: I will fix UTs related with AD batch tasks after all changes ready. I'm ok if @spbjss willing to put some effort to fix it now or leave it to later to avoid duplicate effort.

Comment (kaituo) [reply src/test/java/org/opensearch/ad/transport/ADBatchAnomalyResultTransportActionTests.java:140]: sounds good. Will approve the PR once you get the build issue figured out.  Please let me know.

File: src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java
Comment (dblock) [inline src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: I know @saratvemulapalli has been saying that we would want these to be `plugins.` just to avoid renaming again :) Care to comment?

Comment (saratvemulapalli) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: I think it was miss on our side, we haven't updated the guidance. I would prefer to have `plugins` instead of `opensearch`.
Can we update the doc: https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#settings

Also let folks working on JobScheduler to take care of it as well.

Comment (spbjss) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: Do you mean we should change "opensearch.anomaly_detection.xxx" to "plugins.anomaly_detection.xxx"?

Sure, I will make that change.
Thank you for the comments!

Comment (saratvemulapalli) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: Yes please. Thanks Alex!

Comment (dblock) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: And why would we not make this `anomaly_detection.max_anomaly_detectors`? That's what k-nn is doing, for example.

Comment (saratvemulapalli) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: Definitely that works too. But I would love to have consistency across different pieces in the plugins.
And, we have decided to go with that as we have already discussed in the public:
https://github.com/opensearch-project/opensearch-plugins/issues/13
https://discuss.opendistrocommunity.dev/t/thoughts-on-having-opensearch-identifier-in-all-things-plugins/5855

I am not opposed to either way but since we have taken a decision in the public, I would prefer to stick with it unless we see strong contentions to why that doesnt work. What do you think?

Comment (dblock) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: A-OK by me to include `plugins`. Who's updating the doc? :)

Comment (saratvemulapalli) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java:43]: Haha oncall :/, thanks for the change!

Comment (spbjss) [reply src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java]: Thank you very much for the input!
I have submitted a commits to update "opensearch" to "plugins"

**Comment by ylwu-amzn** [comment]
@spbjss Changed to main branch for OpenSearch in CI (https://github.com/opensearch-project/anomaly-detection/pull/50). Can you merge main branch to check if it can pass the github CI workflow?

**Comment by ylwu-amzn** [comment]
From CI workflow log, the compile can pass, but integTest failed, error message 
```
=== Standard output of node `node{::integTest-0}` ===
> Task :integTest FAILED

»    ↓ errors and warnings from /home/runner/work/anomaly-detection/anomaly-detection/build/testclusters/integTest-0/logs/opensearch.stdout.log ↓
» ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [integTest-0] fatal error in thread [main], exiting
»  java.lang.NoSuchMethodError: 'org.opensearch.common.settings.Setting org.opensearch.common.settings.Setting.positiveTimeSetting(java.lang.String, org.opensearch.common.settings.Setting, org.opensearch.common.settings.Setting$Property[])'
»  	at org.opensearch.ad.settings.AnomalyDetectorSettings.<clinit>(AnomalyDetectorSettings.java:72) ~[?:?]
»  	at org.opensearch.ad.AnomalyDetectorPlugin.getSettings(AnomalyDetectorPlugin.java:639) ~[?:?]
»  	at org.opensearch.plugins.PluginsService.lambda$getPluginSettings$0(PluginsService.java:107) ~[opensearch-1.0.0-beta1.jar:1.0.0-beta1]
»  	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271) ~[?:?]
»  	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?]
```
The integTest will always use `opensearch-1.0.0-beta1` to build test cluster for integTest. @dblock  Any suggestion how to unblock the integTest? Should we build beta2 or 1.x to include the code change of [OpenSearch PR 665](https://github.com/opensearch-project/OpenSearch/pull/665/files#diff-5015a740c24cddcf19da3466776a79f1f8cc3e21527f01261da5e8b782fc1894R1861)?  This PR removed `final TimeValue minValue` from input param list of function `Setting#positiveTimeSetting`. So we can't use beta1 to build test cluster to run latest code.

**Comment by ylwu-amzn** [comment]
Several options to unblock the integTest, 
1. Use old setting function with some redundancy code: https://github.com/ylwu-amzn/anomaly-detection-2/commit/36160805c22571061cca5064257e952d4d7ff4c0.
2. Build new artifact with latest Setting class for integTest. 
3. Ignore the integTest failure, just merge it directly. We may have no way to run integTest until we have a new OpenSearch artifact ready.

**Comment by dblock** [comment]
@ylwu-amzn @spbjss would you please followup with infra team on (1)? cc: @bbarani

**Comment by saratvemulapalli** [comment]
> @ylwu-amzn @spbjss would you please followup with infra team on (1)? cc: @bbarani

Also can we open an issue in opensearch-build to track this?

**Comment by ylwu-amzn** [comment]
> > @ylwu-amzn @spbjss would you please followup with infra team on (1)? cc: @bbarani
> 
> Also can we open an issue in opensearch-build to track this?

Talked with Peter @peterzhuamazon , we will have a new tarball tomorrow. Do we need to change the tarball link used in integTest ? Who will change it? @dblock  @saratvemulapalli

**Comment by dblock** [comment]
> > > @ylwu-amzn @spbjss would you please followup with infra team on (1)? cc: @bbarani
> > 
> > 
> > Also can we open an issue in opensearch-build to track this?
> 
> Talked with Peter @peterzhuamazon , we will have a new tarball tomorrow. Do we need to change the tarball link used in integTest ? Who will change it? @dblock @saratvemulapalli

As a general rule of thumb when we need something we should be opening issues. I got pinged by several people about this, so opened https://github.com/opensearch-project/opensearch-build/issues/39 and just point them to it.

**Comment by dblock** [comment]
> Do we need to change the tarball link used in integTest ? Who will change it? @dblock @saratvemulapalli

@spbjss who opened this PR wants it to get to green so I imagine he would ;)

**Comment by saratvemulapalli** [comment]
> > Do we need to change the tarball link used in integTest ? Who will change it? @dblock @saratvemulapalli
> 
> @spbjss who opened this PR wants it to get to green so I imagine he would ;)

Haha I think the link should work, we have agreed upon a format between Infra and Opensearch, Distribution Downloader should be able to handle that.
https://github.com/opensearch-project/OpenSearch/blob/0e9f74e35f1255cb9ec45be3d8960aad195a9f6e/buildSrc/src/main/java/org/opensearch/gradle/DistributionDownloadPlugin.java#L159

**Comment by spbjss** [comment]
> > > Do we need to change the tarball link used in integTest ? Who will change it? @dblock @saratvemulapalli
> > 
> > 
> > @spbjss who opened this PR wants it to get to green so I imagine he would ;)
> 
> Haha I think the link should work, we have agreed upon a format between Infra and Opensearch, Distribution Downloader should be able to handle that.
> https://github.com/opensearch-project/OpenSearch/blob/0e9f74e35f1255cb9ec45be3d8960aad195a9f6e/buildSrc/src/main/java/org/opensearch/gradle/DistributionDownloadPlugin.java#L159

I could change the tar url of the integration test to "https://artifacts.opensearch.org/releases/core/opensearch/1.0.0-rc1/opensearch-1.0.0-rc1-linux-x64.tar.gz", waiting for infrastructure team to build the rc1 artifact.

**Review by dblock — APPROVED** [review]
Good stuff.

File: .github/workflows/CI.yml
Comment (saratvemulapalli) [inline .github/workflows/CI.yml:36]: Yay!

File: .github/workflows/CI.yml
Comment (saratvemulapalli) [inline .github/workflows/CI.yml:45]: We should add instructions for our branching strategy for plugins in the plugins repo. @dblock 
I can take care of it.

**Review by saratvemulapalli — APPROVED** [review]
Looks great!

File: .github/workflows/CI.yml
Comment (dblock) [reply .github/workflows/CI.yml:45]: Please do.

Comment (saratvemulapalli) [reply .github/workflows/CI.yml:45]: Issue: https://github.com/opensearch-project/opensearch-plugins/issues/35

**Pull request merged**
